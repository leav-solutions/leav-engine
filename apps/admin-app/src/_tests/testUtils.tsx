// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {InMemoryCacheConfig} from '@apollo/client';
import {MockedResponse} from '@apollo/client/testing';
import {render, RenderOptions, RenderResult} from '@testing-library/react';
import ApplicationContext from 'context/CurrentApplicationContext';
import React, {PropsWithChildren, ReactElement} from 'react';
import {MemoryRouterProps} from 'react-router';
import {MemoryRouter} from 'react-router-dom';
import {RootState} from 'redux/store';
import {mockApplicationDetails} from '__mocks__/common/applications';
import MockedLangContextProvider from '__mocks__/MockedLangContextProvider';
import MockedProviderWithFragments from '__mocks__/MockedProviderWithFragments';
import MockedUserContextProvider from '__mocks__/MockedUserContextProvider';
import {MockStore} from '__mocks__/reduxProvider';

interface ICustomRenderOptions extends RenderOptions {
    apolloMocks?: readonly MockedResponse[];
    cacheSettings?: InMemoryCacheConfig;
    routerProps?: MemoryRouterProps;
    storeState?: Partial<RootState>;
    [key: string]: any;
}

interface IProvidersProps {
    apolloMocks?: readonly MockedResponse[];
    cacheSettings?: InMemoryCacheConfig;
    routerProps?: MemoryRouterProps;
    storeState?: Partial<RootState>;
}

const Providers = ({
    children,
    apolloMocks,
    cacheSettings,
    routerProps,
    storeState
}: PropsWithChildren<IProvidersProps>) => {
    return (
        <MockStore state={storeState}>
            <MockedProviderWithFragments mocks={apolloMocks} cacheSettings={cacheSettings}>
                <MockedLangContextProvider>
                    <MockedUserContextProvider>
                        <ApplicationContext.Provider value={mockApplicationDetails}>
                            <MemoryRouter {...routerProps}>{children as ReactElement}</MemoryRouter>
                        </ApplicationContext.Provider>
                    </MockedUserContextProvider>
                </MockedLangContextProvider>
            </MockedProviderWithFragments>
        </MockStore>
    );
};

// Wrapper around testing-library's render to automatically render apollo's provider and redux store provider
const renderWithProviders = (ui: ReactElement, options?: ICustomRenderOptions): RenderResult =>
    render(ui, {wrapper: props => <Providers {...props} {...options} />, ...options});

// Re-export everything from testing-library to improve DX. You can everything you need from this file when you use this
// custom render
export * from '@testing-library/react';
export {renderWithProviders as render};
