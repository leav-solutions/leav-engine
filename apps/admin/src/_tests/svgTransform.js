// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
module.exports = {
    process() {
        return {code: 'module.exports = {};'};
    },
    getCacheKey() {
        // The output is always the same.
        return 'svgTransform';
    }
};
