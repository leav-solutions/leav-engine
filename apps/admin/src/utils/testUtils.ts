// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
export async function wait(duration = 0) {
    return new Promise(resolve => setTimeout(resolve, duration));
}
