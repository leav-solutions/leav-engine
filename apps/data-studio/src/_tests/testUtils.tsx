// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {InMemoryCacheConfig} from '@apollo/client';
import {MockedResponse} from '@apollo/client/testing';
import {render, RenderOptions, RenderResult} from '@testing-library/react';
import ApplicationContext from 'context/ApplicationContext';
import {IApplicationContext} from 'context/ApplicationContext/_types';
import {PropsWithChildren, ReactElement} from 'react';
import {RootState} from 'redux/store';
import {GET_APPLICATION_BY_ID_applications_list} from '_gqlTypes/GET_APPLICATION_BY_ID';
import {GET_GLOBAL_SETTINGS_globalSettings} from '_gqlTypes/GET_GLOBAL_SETTINGS';
import {mockApplicationDetails} from '__mocks__/common/applications';
import MockStore from '__mocks__/common/mockRedux/mockStore';
import MockedProviderWithFragments from '__mocks__/MockedProviderWithFragments';

export interface ICustomRenderOptions extends RenderOptions {
    apolloMocks?: readonly MockedResponse[];
    storeState?: Partial<RootState>;
    cacheSettings?: InMemoryCacheConfig;
    currentApp?: GET_APPLICATION_BY_ID_applications_list;
    [key: string]: any;
}

interface IProvidersProps {
    apolloMocks?: readonly MockedResponse[];
    storeState?: Partial<RootState>;
    cacheSettings?: InMemoryCacheConfig;
    currentApp?: GET_APPLICATION_BY_ID_applications_list;
    globalSettings?: GET_GLOBAL_SETTINGS_globalSettings;
}

const Providers = ({
    children,
    apolloMocks,
    storeState,
    cacheSettings,
    currentApp,
    globalSettings
}: PropsWithChildren<IProvidersProps>) => {
    const appContextData: IApplicationContext = {
        currentApp: currentApp ?? mockApplicationDetails,
        globalSettings: {
            name: 'My App',
            icon: null,
            ...globalSettings
        }
    };

    return (
        <MockedProviderWithFragments mocks={apolloMocks} cacheSettings={cacheSettings}>
            <MockStore state={storeState}>
                <ApplicationContext.Provider value={appContextData}>{children ?? <></>}</ApplicationContext.Provider>
            </MockStore>
        </MockedProviderWithFragments>
    );
};

// Wrapper around testing-library's render to automatically render apollo's provider and redux store provider
const renderWithProviders = (ui: ReactElement, options?: ICustomRenderOptions): RenderResult =>
    render(ui, {wrapper: props => <Providers {...props} {...options} />, ...options});

// Re-export everything from testing-library to improve DX. You can everything you need from this file when you use this
// custom render
export * from '@testing-library/react';
export {renderWithProviders as render};
