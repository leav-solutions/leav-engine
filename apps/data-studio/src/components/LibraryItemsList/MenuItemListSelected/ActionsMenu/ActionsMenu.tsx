// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {DeleteOutlined, DownOutlined, ExportOutlined} from '@ant-design/icons';
import {Button, Dropdown, Menu} from 'antd';
import {ReactNode, useState} from 'react';
import {useTranslation} from 'react-i18next';
import DeactivateRecordsModal from '../DeactivateRecordsModal';
import ExportModal from './ExportModal';

interface IMenuAction {
    key: string;
    icon: ReactNode;
    title: string;
    modalComp: (props: any) => JSX.Element;
}

function ActionsMenu(): JSX.Element {
    const {t} = useTranslation();
    const [activeAction, setActiveAction] = useState<string>();

    const actions: IMenuAction[] = [
        {
            key: 'export',
            icon: <ExportOutlined rotate={270} />,
            title: t('export.title'),
            modalComp: ExportModal
        },
        {
            key: 'deactivate',
            icon: <DeleteOutlined />,
            title: t('records_deactivation.title'),
            modalComp: DeactivateRecordsModal
        }
    ];

    const _handleClick = (actionKey: string) => () => {
        setActiveAction(actionKey);
    };

    const _handleCloseModal = () => setActiveAction('');

    return (
        <>
            <Dropdown
                overlay={
                    <Menu
                        items={actions.map(a => ({
                            icon: a.icon,
                            key: a.key,
                            title: a.title,
                            label: a.title,
                            onClick: _handleClick(a.key)
                        }))}
                    ></Menu>
                }
            >
                <Button>
                    {t('menu-selection.actions')} <DownOutlined />
                </Button>
            </Dropdown>
            {actions
                .filter(a => activeAction === a.key)
                .map(a => (
                    <a.modalComp key={a.key} open={activeAction === a.key} onClose={_handleCloseModal} />
                ))}
        </>
    );
}

export default ActionsMenu;
