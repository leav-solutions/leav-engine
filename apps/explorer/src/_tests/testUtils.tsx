// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {InMemoryCacheConfig} from '@apollo/client';
import {MockedResponse} from '@apollo/client/testing';
import {render, RenderOptions, RenderResult} from '@testing-library/react';
import ApplicationContext from 'context/ApplicationContext';
import {PropsWithChildren, ReactElement} from 'react';
import {RootState} from 'redux/store';
import {GET_APPLICATION_BY_ID_applications_list} from '_gqlTypes/GET_APPLICATION_BY_ID';
import {mockApplicationDetails} from '__mocks__/common/applications';
import MockStore from '__mocks__/common/mockRedux/mockStore';
import MockedProviderWithFragments from '__mocks__/MockedProviderWithFragments';

export interface ICustomRenderOptions extends RenderOptions {
    apolloMocks?: readonly MockedResponse[];
    storeState?: Partial<RootState>;
    cacheSettings?: InMemoryCacheConfig;
    currentApp?: GET_APPLICATION_BY_ID_applications_list;
    [key: string]: any;
}

interface IProvidersProps {
    apolloMocks?: readonly MockedResponse[];
    storeState?: Partial<RootState>;
    cacheSettings?: InMemoryCacheConfig;
    currentApp?: GET_APPLICATION_BY_ID_applications_list;
}

const Providers = ({
    children,
    apolloMocks,
    storeState,
    cacheSettings,
    currentApp
}: PropsWithChildren<IProvidersProps>) => {
    return (
        <MockedProviderWithFragments mocks={apolloMocks} cacheSettings={cacheSettings}>
            <MockStore state={storeState}>
                <ApplicationContext.Provider value={currentApp ?? mockApplicationDetails}>
                    {children ?? <></>}
                </ApplicationContext.Provider>
            </MockStore>
        </MockedProviderWithFragments>
    );
};

// Wrapper around testing-library's render to automatically render apollo's provider and redux store provider
const renderWithProviders = (ui: ReactElement, options?: ICustomRenderOptions): RenderResult =>
    render(ui, {wrapper: props => <Providers {...props} {...options} />, ...options});

// Re-export everything from testing-library to improve DX. You can everything you need from this file when you use this
// custom render
export * from '@testing-library/react';
export {renderWithProviders as render};
