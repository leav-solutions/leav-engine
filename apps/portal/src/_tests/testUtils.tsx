// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {InMemoryCache, InMemoryCacheConfig} from '@apollo/client';
import {MockedProvider, MockedResponse} from '@apollo/client/testing';
import {MockedLangContextProvider} from '@leav/ui';
import {render, RenderOptions, RenderResult} from '@testing-library/react';
import React, {PropsWithChildren, ReactElement} from 'react';

interface ICustomRenderOptions extends RenderOptions {
    apolloMocks?: readonly MockedResponse[];
    cacheSettings?: InMemoryCacheConfig;
    [key: string]: any;
}

interface IProvidersProps {
    apolloMocks?: readonly MockedResponse[];
    cacheSettings?: InMemoryCacheConfig;
}

const Providers = ({children, apolloMocks, cacheSettings}: PropsWithChildren<IProvidersProps>) => {
    const mockCache = new InMemoryCache(cacheSettings);
    return (
        <MockedLangContextProvider>
            <MockedProvider mocks={apolloMocks} cache={mockCache}>
                {children ?? <></>}
            </MockedProvider>
        </MockedLangContextProvider>
    );
};

// Wrapper around testing-library's render to automatically render apollo's provider and redux store provider
const renderWithProviders = (ui: ReactElement, options?: ICustomRenderOptions): RenderResult =>
    render(ui, {wrapper: props => <Providers {...props} {...options} />, ...options});

// Re-export everything from testing-library to improve DX. You can everything you need from this file when you use this
// custom render
export * from '@testing-library/react';
export {renderWithProviders as render};
