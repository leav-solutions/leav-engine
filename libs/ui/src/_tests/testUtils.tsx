// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import {InMemoryCache, InMemoryCacheConfig} from '@apollo/client';
import {MockedProvider, MockedResponse} from '@apollo/client/testing';
import {render, RenderOptions, RenderResult} from '@testing-library/react';
import {PropsWithChildren, ReactElement} from 'react';
import MockedLangContextProvider from '../testing/MockedLangContextProvider';

interface ICustomRenderOptions extends RenderOptions {
    mocks?: readonly MockedResponse[];
    [key: string]: any;
}

interface IProvidersProps {
    mocks?: readonly MockedResponse[];
    cacheSettings?: InMemoryCacheConfig;
}

const Providers = ({children, mocks, cacheSettings}: PropsWithChildren<IProvidersProps>) => {
    const mockCache = new InMemoryCache(cacheSettings);

    return (
        <MockedLangContextProvider>
            <MockedProvider mocks={mocks} cache={mockCache} addTypename>
                {children ?? <></>}
            </MockedProvider>
        </MockedLangContextProvider>
    );
};

// Wrapper around testing-library's render to automatically render apollo's provider and redux store provider
const renderWithProviders = (ui: ReactElement, options?: ICustomRenderOptions): RenderResult => {
    return render(ui, {wrapper: props => <Providers {...props} {...options} />, ...options});
};

// Re-export everything from testing-library to improve DX. You can everything you need from this file when you use this
// custom render
export * from '@testing-library/react';
export {renderWithProviders as render};
