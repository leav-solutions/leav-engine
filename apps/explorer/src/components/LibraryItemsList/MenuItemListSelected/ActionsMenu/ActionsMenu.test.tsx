// Copyright LEAV Solutions 2017
// This file is released under LGPL V3
// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
import userEvent from '@testing-library/user-event';
import React from 'react';
import {act, render, screen} from '_tests/testUtils';
import ActionsMenu from './ActionsMenu';

jest.mock('./ExportModal', () => {
    return function ExportModal() {
        return <div>ExportModal</div>;
    };
});

describe('ActionsMenu', () => {
    test('Render menu', async () => {
        render(<ActionsMenu />);

        expect(screen.getByRole('button')).toBeInTheDocument();

        await act(async () => {
            userEvent.click(screen.getByRole('button'));
        });

        expect(await screen.findByText(/export/)).toBeInTheDocument();
    });
});
