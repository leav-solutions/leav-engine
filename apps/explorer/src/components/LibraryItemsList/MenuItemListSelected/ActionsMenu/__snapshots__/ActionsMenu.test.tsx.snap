// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionsMenu Snapshot test 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": "ant-btn ant-btn-default ant-dropdown-trigger",
      "type": "button",
    },
    "children": Array [
      Node {
        "attribs": Object {},
        "children": Array [
          Node {
            "data": "menu-selection.actions ",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "aria-label": "down",
            "class": "anticon anticon-down",
            "role": "img",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "aria-hidden": "true",
                "data-icon": "down",
                "fill": "currentColor",
                "focusable": "false",
                "height": "1em",
                "viewBox": "64 64 896 896",
                "width": "1em",
              },
              "children": Array [
                Node {
                  "attribs": Object {
                    "d": "M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z",
                  },
                  "children": Array [],
                  "name": "path",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "d": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "d": undefined,
                  },
                },
              ],
              "name": "svg",
              "namespace": "http://www.w3.org/2000/svg",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-hidden": undefined,
                "data-icon": undefined,
                "fill": undefined,
                "focusable": undefined,
                "height": undefined,
                "viewBox": undefined,
                "width": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-hidden": undefined,
                "data-icon": undefined,
                "fill": undefined,
                "focusable": undefined,
                "height": undefined,
                "viewBox": undefined,
                "width": undefined,
              },
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "aria-label": undefined,
            "class": undefined,
            "role": undefined,
          },
          "x-attribsPrefix": Object {
            "aria-label": undefined,
            "class": undefined,
            "role": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      Node {
        "attribs": Object {
          "aria-label": "down",
          "class": "anticon anticon-down",
          "role": "img",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "aria-hidden": "true",
              "data-icon": "down",
              "fill": "currentColor",
              "focusable": "false",
              "height": "1em",
              "viewBox": "64 64 896 896",
              "width": "1em",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "d": "M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z",
                },
                "children": Array [],
                "name": "path",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "d": undefined,
                },
                "x-attribsPrefix": Object {
                  "d": undefined,
                },
              },
            ],
            "name": "svg",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "aria-hidden": undefined,
              "data-icon": undefined,
              "fill": undefined,
              "focusable": undefined,
              "height": undefined,
              "viewBox": undefined,
              "width": undefined,
            },
            "x-attribsPrefix": Object {
              "aria-hidden": undefined,
              "data-icon": undefined,
              "fill": undefined,
              "focusable": undefined,
              "height": undefined,
              "viewBox": undefined,
              "width": undefined,
            },
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "data": "menu-selection.actions ",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "aria-label": undefined,
          "class": undefined,
          "role": undefined,
        },
        "x-attribsPrefix": Object {
          "aria-label": undefined,
          "class": undefined,
          "role": undefined,
        },
      },
    ],
    "name": "button",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "type": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "type": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
